### DISCUSSION TITLE
How to send J1939 error indicators

### DISCUSSION THREAD

**Discussion by euttsov** [cantools/cantools#449] [Q&A] [answered]
Hi all! I need to send NA values for several J1939 signals from message. NA indicator is signal values with all bits set to 1. 
I cannot find a way how to encode some signals without scaling and some with scaling and with correct values.
Could somebody give me a direction?

**Comment by andlaus** [comment]
your signals in question ought to exhibit named values where all bits to are set. You can specify these values to be used by passing a string for the signal's value for `encode()`:

```python
import cantools

db = cantools.database.load_file('motohawk.dbc')
msg = db.get_message_by_name('ExampleMessage')
raw_data = message.encode({
  'Enable': 'Enabled',
  'AverageRadius': 'NotAvailable', # <- all bits 1, needs to be added to the DBC file
  'Temperature': 'NotAvailable', # <- same as above
})
```

Possibly, a more generic mechanism will be added in the future, but in my experience this works rather well..

**Reply by euttsov** [reply]
This could be not only enum values, but also integer ranges. And unfortunately i cannot control DBC files. Need to find another way.

**Comment by andlaus** [comment, marked as answer]
as far as I know it is possible to define some choices besides ranged values. I'm not an expert in DBC, though. (ARXML definitely allows this.)

**Reply by euttsov** [reply]
Yes, you give me idea. I will try to calculate and append 'NotAvailable' value to signal choices before encoding message. Thank you!

**Reply by euttsov** [reply]
But signal choices is readonly property. And again - DBC allows to define values table for signal, but i can't change them.

**Reply by euttsov** [reply]
Finally it works by overriding protected field _choices. Thanks.
